Sources/App/routes+documentation.swift

```diff
@@ -161,11 +161,13 @@ private extension Request {
 
         let docVersion = try await { () -> DocRoute.DocVersion in
             if reference == String.current {
-                guard let params = try await DocumentationTarget.query(on: db, owner: owner, repository: repository)?.internal
-                else { throw Abort(.notFound) }
-                if fragment.requiresArchive && archive?.lowercased() != params.archive.lowercased() {
+                if let ref = Current.currentReferenceCache()?[owner: owner, repository: repository] {
+                    return .current(referencing: ref)
+                }
+                guard let params = try await DocumentationTarget.query(on: db, owner: owner, repository: repository)?.internal else {
                     throw Abort(.notFound)
                 }
+                Current.currentReferenceCache()?[owner: owner, repository: repository] = params.reference
                 return .current(referencing: params.reference)
```

Comment:
Once we've looked up the current reference for a given owner/repo we keep it around in the cache for 5 minutes. That way we don't need to fiddle with any further rewrites (we can keep the resources in the HTML as ~ refs - might as well, since the JS makes calls to ~ resources anyway) and can just rely on this cache.

Reply:
It's 19 db requests on initial documentation page load. I believe subsequent loads do not hit the db but even so we really don't want all those unnecessary requests hitting the db.